diff -r b1a7fae2cf18 configure.in
--- a/configure.in	Fri Jan 11 22:20:21 2013 +0000
+++ b/configure.in	Mon Jan 28 16:46:08 2013 +0100
@@ -8773,18 +8773,22 @@ if test -n "$MOZ_WEBRTC"; then
      ${srcdir}/media/webrtc/trunk/testing/gtest.gyp
    if test "$?" != 0; then
       AC_MSG_ERROR([failed to generate gtest Makefiles])
    fi
 fi
 
 # Generate a JSON config file for unittest harnesses etc to read
 # build configuration details from in a standardized way.
-OS_TARGET=${OS_TARGET} TARGET_CPU=${TARGET_CPU} MOZ_DEBUG=${MOZ_DEBUG} \
-MOZ_WIDGET_TOOLKIT=${MOZ_WIDGET_TOOLKIT} UNIVERSAL_BINARY=${UNIVERSAL_BINARY} \
+OS_TARGET=${OS_TARGET} \
+TARGET_CPU=${TARGET_CPU} \
+MOZ_DEBUG=${MOZ_DEBUG} \
+MOZ_WIDGET_TOOLKIT=${MOZ_WIDGET_TOOLKIT} \
+UNIVERSAL_BINARY=${UNIVERSAL_BINARY} \
+MOZ_CRASHREPORTER=${MOZ_CRASHREPORTER} \
   $PYTHON ${_topsrcdir}/config/writemozinfo.py ./mozinfo.json.tmp
 if cmp -s ./mozinfo.json.tmp ./mozinfo.json; then
   rm ./mozinfo.json.tmp
 else
   mv -f ./mozinfo.json.tmp ./mozinfo.json
 fi
 
 # Run jemalloc configure script
diff -r b1a7fae2cf18 testing/xpcshell/xpcshell.ini
--- a/testing/xpcshell/xpcshell.ini	Fri Jan 11 22:20:21 2013 +0000
+++ b/testing/xpcshell/xpcshell.ini	Mon Jan 28 16:46:08 2013 +0100
@@ -95,20 +95,20 @@ skip-if = os == "android"
 [include:extensions/spellcheck/hunspell/tests/unit/xpcshell.ini]
 [include:toolkit/components/search/tests/xpcshell/xpcshell.ini]
 [include:toolkit/mozapps/shared/test/unit/xpcshell.ini]
 [include:services/crypto/component/tests/unit/xpcshell.ini]
 [include:layout/tools/layout-debug/tests/unit/xpcshell.ini]
 skip-if = !debug
 
 [include:toolkit/crashreporter/test/unit/xpcshell.ini]
-skip-if = os == "linux" || !crashreporter
+skip-if = !crashreporter
 
 [include:toolkit/crashreporter/test/unit_ipc/xpcshell.ini]
-skip-if.os == "linux" || !crashreporter
+skip-if = !crashreporter
 
 #XXX: we don't actually set os = maemo
 [include:toolkit/crashreporter/client/maemo-unit/xpcshell.ini]
 run-if = os == "maemo"
 
 [include:toolkit/components/commandlines/test/unit_win/xpcshell.ini]
 skip-if = os != "win"
 
diff -r b1a7fae2cf18 toolkit/crashreporter/test/CrashTestUtils.jsm
--- a/toolkit/crashreporter/test/CrashTestUtils.jsm	Fri Jan 11 22:20:21 2013 +0000
+++ b/toolkit/crashreporter/test/CrashTestUtils.jsm	Mon Jan 28 16:46:08 2013 +0100
@@ -17,18 +17,19 @@ let CrashTestUtils = {
 
   // Constants for dumpHasStream()
   // From google_breakpad/common/minidump_format.h
   MD_THREAD_LIST_STREAM:       3,
   MD_MEMORY_INFO_LIST_STREAM:  16
 };
 
 // Grab APIs from the testcrasher shared library
+Components.utils.import("resource://gre/modules/Services.jsm");
 Components.utils.import("resource://gre/modules/ctypes.jsm");
-let dir = __LOCATION__.parent;
+let dir = Services.dirsvc.get("CurWorkD", Components.interfaces.nsILocalFile);
 let file = dir.clone();
 file.append(ctypes.libraryName("testcrasher"));
 let lib = ctypes.open(file.path);
 CrashTestUtils.crash = lib.declare("Crash",
                                    ctypes.default_abi,
                                    ctypes.void_t,
                                    ctypes.int16_t);
 CrashTestUtils.saveAppMemory = lib.declare("SaveAppMemory",
diff -r b1a7fae2cf18 toolkit/crashreporter/test/unit/xpcshell.ini
--- a/toolkit/crashreporter/test/unit/xpcshell.ini	Fri Jan 11 22:20:21 2013 +0000
+++ b/toolkit/crashreporter/test/unit/xpcshell.ini	Mon Jan 28 16:46:08 2013 +0100
@@ -1,14 +1,16 @@
 [DEFAULT]
 head = head_crashreporter.js
 tail = 
 
 [test_crash_moz_crash.js]
 [test_crash_purevirtual.js]
 [test_crash_runtimeabort.js]
 [test_crash_oom.js]
+skip-if = os == 'win' && debug
+
 [test_crashreporter.js]
 [test_crashreporter_crash.js]
 [test_crashreporter_crash_profile_lock.js]
 
 [test_crashreporter_appmem.js]
 run-if = os == 'win' || os == 'linux'
